staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/types.go

```diff
@@ -520,36 +533,38 @@ var (
 	AnyType = newSimpleType("any", decls.Any, nil)
 
 	// BoolType is equivalent to the CEL 'bool' type.
-	BoolType = newSimpleType("bool", decls.Bool, types.False)
+	BoolType = newSimpleTypeWithMinSize("bool", decls.Bool, types.False, minBoolSize)
 
 	// BytesType is equivalent to the CEL 'bytes' type.
-	BytesType = newSimpleType("bytes", decls.Bytes, types.Bytes([]byte{}))
+	BytesType = newSimpleTypeWithMinSize("bytes", decls.Bytes, types.Bytes([]byte{}), minStringSize)
 
 	// DoubleType is equivalent to the CEL 'double' type which is a 64-bit floating point value.
-	DoubleType = newSimpleType("double", decls.Double, types.Double(0))
+	DoubleType = newSimpleTypeWithMinSize("double", decls.Double, types.Double(0), minNumberSize)
 
 	// DurationType is equivalent to the CEL 'duration' type.
-	DurationType = newSimpleType("duration", decls.Duration, types.Duration{Duration: time.Duration(0)})
+	DurationType = newSimpleTypeWithMinSize("duration", decls.Duration, types.Duration{Duration: time.Duration(0)}, minDurationSizeJSON)
 
 	// DateType is equivalent to the CEL 'date' type.
-	DateType = newSimpleType("date", decls.Timestamp, types.Timestamp{Time: time.Time{}})
+	DateType = newSimpleTypeWithMinSize("date", decls.Timestamp, types.Timestamp{Time: time.Time{}}, dateSizeJSON)
 
 	// DynType is the equivalent of the CEL 'dyn' concept which indicates that the type will be
 	// determined at runtime rather than compile time.
 	DynType = newSimpleType("dyn", decls.Dyn, nil)
 
 	// IntType is equivalent to the CEL 'int' type which is a 64-bit signed int.
-	IntType = newSimpleType("int", decls.Int, types.IntZero)
+	IntType = newSimpleTypeWithMinSize("int", decls.Int, types.IntZero, minNumberSize)
 
 	// NullType is equivalent to the CEL 'null_type'.
 	NullType = newSimpleType("null_type", decls.Null, types.NullValue)
```

Comment:
size of 4?

Reply:
Since the minimum size refers to how small the type can be encoded in JSON in bytes, shouldn't it be 1 since character `0` takes up 1 byte?

Reply:
oh, I thought this meant a literal `null`

Reply:
Whoops, I thought this comment was referring to `IntType` for some reason. I'm uncertain what the minimum size for `NullType` is; `SchemaDeclType` doesn't have a case for it so I'm not actually sure how anything could have `NullType` at the moment. `null`/4 makes a lot of sense, though.

Reply:
(resolving this since #111328 tracks `NullType` and possibly `UIntType` removal, feel free to reopen if you think it'd be better off solved in this PR)